protos/flyteidl/core/workflow.proto

```diff
@@ -124,6 +127,8 @@ message Node {
 // Metadata for the entire workflow.
 // To be used in the future.
 message WorkflowMetadata {
+    // Identify whether workflow is interruptible
```

Comment:
A couple of things:
- In the generated golang proto, .GetInterruptible() will return false whether it's explicitly set to false or not set at all. Knowing this, how will this flag interact with the per node flag?
- After resolving this ^, please add to the comment here how precedence will work...

Reply:
So the behavior should be whatever value is set at the workflow level than all nodes should be set to that value as well unless a value has already been set at the node level. In this case I believe using a `oneof` value should be the correct message type instead of bool. @EngHabu would you agree?

Reply:
We still need a default behavior for the workflow. What do we do if interruptible isn't defined? Yes, the node will take precedence.

Reply:
@surindersinghp I'd propose for the moment if not set than we should default to **not** interruptible

Reply:
then bool should be good, right? default=false.

Reply:
@surindersinghp if my understanding is correct bool will not work because even if the value is not set .GetInterruptible() will return false. If we have the value set to true at the workflow level there isn't a way to distinguish between a user setting interruptible to false or that value not being set at all at the node level.
Using a `oneof` we can check if the value is not set at the node level then use the value set at the workflow level (with default being false if it isn't set).

Reply:
sure, I meant at the workflow level (WorkflowMetadata). Yes, for node level it should be oneof